src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -150,13 +143,17 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
 
                 // Add any additional common System parameters
                 // Add ExecutionContext to provide access to InvocationId, etc.
-                descriptors.Add(new ParameterDescriptor("context", typeof(ExecutionContext)));
-                
-                // If we have an HTTP trigger binding but we're not binding
-                // to the HttpRequestMessage, require it as a system parameter
-                if (addHttpRequestSystemParameter)
+                if (!descriptors.Any(p => p.Name == ScriptConstants.SystemExecutionContextParameterName && p.Type == typeof(ExecutionContext)))
```

Comment:
It's a TODO for me to make the same change for the ExecutionContext system parameter - if the user defines one we use it, otherwise we define our own.